python/ray/serve/controller.py

```diff
@@ -759,7 +767,11 @@ async def create_backend(self, backend_tag, backend_config,
                     backend_tag] = BasicAutoscalingPolicy(
                         backend_tag, backend_config.autoscaling_config)
 
-            self._scale_replicas(backend_tag, backend_config.num_replicas)
+            try:
+                self._scale_replicas(backend_tag, backend_config.num_replicas)
+            except RayServeException as e:
+                del self.backends[backend_tag]
+                raise e
```

Comment:
Good catch! Do we need similar logic in other API call methods?

Reply:
Thanks, the fix is due to Simon (https://anyscaleteam.slack.com/archives/C011RJWAV08/p1596670952010100).  I don't think we need similar logic elsewhere but I'm not very confident about that, so I'll check with Simon.